packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -716,9 +717,13 @@ class IOSDeviceLogReader extends DeviceLogReader {
     }
     // Add the debugger logs to the controller created on initialization.
     _loggingSubscriptions.add(debugger.logLines.listen(
-      (String line) => _linesController.add(_debuggerLineHandler(line)),
-      onError: _linesController.addError,
-      onDone: _linesController.close,
+      (String line) {
+        if (!linesController.isClosed) {
```

Comment:
This is the actual fix, the other added if checks are just defensive.

Reply:
Why here and not in e.g. `_newSyslogLineHandler()`, or up on line 697?

Reply:
Were there multiple stacktraces we're seeing in CI, or is this a more general question? The stacktraces I saw where all on line 719.

Should I wrap the StreamController so that every `.add()` call has a check first if the delegate is closed?

Reply:
Or I can add a new getter on the `IOSDeviceLogReader` class

Reply:
Done